tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java

```diff
@@ -181,15 +176,23 @@ private ApplicationMaster() {
     hdfsConf = new Configuration(false);
     yarnConf = new Configuration(false);
 
-    hbMonitor = new AbstractLivelinessMonitor<TonyTask>("Tony Task liveliness Monitor") {
+    hbMonitor = new AbstractLivelinessMonitor<TonyTask>("Tony Task liveliness Monitor", new UTCClock()) {
       @Override
       protected void expire(TonyTask task) {
         onTaskDeemedDead(task);
       }
 
       @Override
       protected void serviceStart() throws Exception {
-        setMonitorInterval(hbInterval * 3);
+        // setMonitorInterval(int) changed to setMonitorInterval(long) in Hadoop 2.9,
```

Comment:
This works for both 2.7 and 2.9 because it uses reflection to iterate over all methods in `AbstractLivelinessMonitor`. In 2.7, it will find `setMonitorInterval(int)`, and in 2.9, it will find `setMonitorInterval(long)`. Java will do automatic casting of int to long if the 2.9 `setMonitorInterval(long)` method is found.